### PR TITLE
#457 valet key pattern implement

### PR DESCRIPTION
**Pull request by EdisonE3**
#457 valet key pattern

- Implement valet key:   
Simulate user requests a source from target database. The user access the resource directly by using a token generated by application. 

- Pull request description  
some classes and tests

### CODE REVIEW

File: valet-key/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+layout: pattern
+title: Valet Key
+folder: valet-key
+permalink: /patterns/valet-key/
+categories: Behavioral
+tags:
+ - Data access
```
Comment (iluwatar) [inline valet-key/README.md:8]: Valet key is a cloud pattern and I'd preferably like to see it implemented using cloud technologies. One cloud pattern example is at https://github.com/iluwatar/java-design-patterns/tree/master/cloud-static-content-hosting. 

In the example implementation this article series might also be interesting: https://www.keepsecure.ca/blog/aws-vs-azure-object-storage-valet-key-pattern-part-1/

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: valet-key/README.md (hunk 1 above)
Comment (EdisonE3) [reply valet-key/README.md:8]: Thank for your advice. But it is my first time to use cloud technology, I may take more to finish it.

Comment (iluwatar) [reply valet-key/README.md:8]: No problem @EdisonE3, take your time. I think learning cloud technologies is an excellent investment in the future.

Comment (EdisonE3) [reply valet-key/README.md]: Can I use a local server to replace a real cloud server? I think use local server will be more easy to be tested and if there is necessary, users can deploy it to the Cloud by themselves. @iluwatar

Comment (iluwatar) [reply valet-key/README.md]: @EdisonE3 cloud patterns need to be implemented using cloud technologies. See https://github.com/iluwatar/java-design-patterns/wiki/10.-Technology-selections

**Comment by iluwatar** [comment]
Closed due to inactivity

**Pull request closed without merging**